node/network/availability-distribution/src/tests/mock.rs

```diff
@@ -22,10 +22,13 @@ use sp_keyring::Sr25519Keyring;
 
 use polkadot_erasure_coding::{branches, obtain_chunks_v1 as obtain_chunks};
 use polkadot_node_primitives::{AvailableData, BlockData, ErasureChunk, PoV, Proof};
-use polkadot_primitives::v2::{
-	CandidateCommitments, CandidateDescriptor, CandidateHash, CommittedCandidateReceipt,
-	GroupIndex, Hash, HeadData, Id as ParaId, IndexedVec, OccupiedCore, PersistedValidationData,
-	SessionInfo, ValidatorIndex,
+use polkadot_primitives::{
+	v2::{
```

Comment:
Given that those don't change, it likely makes more sense to move everything over to v3 and just keep the old `SessionInfo` in v2.

Reply:
And what to do with the encapsulated primitives?

For example, `SessionInfo` encapsulates `ValidatorIndex`, which is also defined in primitives. Should I leave `ValidatorIndex` in v2 and make `v3::SessionInfo` use `v2::ValidatorIndex`, or, vice-versa, move `ValidatorIndex` to v3 and make `v2::SessionInfo` use `v3::ValidatorIndex`?

CC @tdimitrov

Reply:
Well either the last option, which favors "We don't care about historic craft." or alternatively, if we introduce a new version, we just re-export everything there from the previous version that still exists in the new version and keep on building stuff on top. While this would be a good history lesson, it would also make it harder to find the actual definitions of things and in general will become unwieldy. Hence yeah, I would go for just using v3 stuff in v2::SessionInfo.

Reply:
Okay, I'm going to participate in a challenge for the longest Polkadot PR with this change someday. But, nevertheless, here it is.

Reply:
Raised issue #6480 on that.